skrub/_agg_joiner.py

```diff
@@ -263,17 +296,24 @@ def fit_transform(self, X, y=None):
         DataFrame
             The augmented input.
         """
-        self._check_inputs(X)
+        if isinstance(self.aux_table, str) and self.aux_table == "X":
+            self.aux_table = X
+        elif not sbd.is_dataframe(self.aux_table):
+            raise ValueError(
+                "'aux_table' must be a dataframe or the string 'X', got"
+                f" {type(self.aux_table)}. If you have more than one 'aux_table',"
+                " use the MultiAggJoiner instead."
+            )
+        self._aux_table = CheckInputDataFrame().fit_transform(self.aux_table)
         self._main_check_input = CheckInputDataFrame()
         X = self._main_check_input.fit_transform(X)
+        self._check_inputs(X)
 
-        skrub_px, _ = get_df_namespace(self._aux_table)
-        self.aux_table_ = skrub_px.aggregate(
+        self.aux_table_ = aggregate(
```

Comment:
as it is of a different nature than `aux_table` (which is not aggregated) maybe it should be given a more different name eg `aggregated_aux_table`